operator/api/redpanda/v1alpha3/redpanda_conversion.go

```diff
@@ -0,0 +1,954 @@
+		// Unsupported on the internal listener in v1alpha2.
+		AdvertisedPorts: nil,
+		PrefixTemplate:  nil,
+	}
+
+	i := 0
+	for _, name := range slices.Sorted(maps.Keys(cfg.External)) {
+		l := cfg.External[name]
+
+		if !l.IsEnabled() {
+			continue
+		}
+
+		i++
+
+		var listenerTLS *ListenerTLS
+		if l.TLS != nil && l.TLS.IsEnabled(&cfg.TLS, tls) {
+			// InternalTLS and ExternalTLS are effectively the same type sans
+			// optionality of fields. To dry this up, we just upcast external
+			// to internal.
+			listenerTLS = convertInternalTLSToListenerTLS(tls, redpanda.InternalTLS{
+				Enabled:           ptr.To(true),
+				Cert:              ptr.Deref(l.TLS.Cert, ""),
+				RequireClientAuth: ptr.Deref(l.TLS.RequireClientAuth, false),
+				TrustStore:        l.TLS.TrustStore,
+			})
+		}
+
+		listeners[i] = Listener{
+			Name:                 name,
+			Port:                 l.Port,
+			AdvertisedPorts:      l.AdvertisedPorts,
+			PrefixTemplate:       l.PrefixTemplate,
+			AuthenticationMethod: (*AuthenticationMethod)(unsafe.Pointer(l.AuthenticationMethod)),
+			TLS:                  listenerTLS,
+		}
+	}
+
+	return listeners[:i+1]
+}
+
+func convertListenerTLSToCertificate(tls *ListenerTLS) *redpandav1alpha2.Certificate {
+	if tls.IssuerRef != nil {
+		return &redpandav1alpha2.Certificate{
+			Enabled: ptr.To(true),
+			IssuerRef: &redpandav1alpha2.IssuerRef{
+				Name:  ptr.To(tls.IssuerRef.Name),
+				Kind:  (*string)(ptr.To(tls.IssuerRef.Kind)),
+				Group: nil,
+			},
+			ApplyInternalDNSNames: nil,
+			CAEnabled:             nil,
+			ClientSecretRef:       nil,
+			Duration:              nil,
+			SecretRef:             nil,
+		}
+	}
+
+	if tls.Secrets != nil {
+		var clientRef *redpandav1alpha2.SecretRef
+		if tls.Secrets.ClientSecretRef != nil {
+			clientRef = &redpandav1alpha2.SecretRef{
+				Name: ptr.To(tls.Secrets.ClientSecretRef.Name),
+			}
+		}
+
+		return &redpandav1alpha2.Certificate{
+			Enabled: ptr.To(true),
+			SecretRef: &redpandav1alpha2.SecretRef{
+				Name: ptr.To(tls.Secrets.SecretRef.Name),
+			},
+			ApplyInternalDNSNames: nil,
+			CAEnabled:             nil,
+			ClientSecretRef:       clientRef,
+			Duration:              nil,
+			IssuerRef:             nil,
+		}
+	}
+
+	return &redpandav1alpha2.Certificate{
+		Enabled:               ptr.To(true),
+		Duration:              tls.CertManager.Duration,
+		ApplyInternalDNSNames: nil,
+		CAEnabled:             nil,
+		ClientSecretRef:       nil,
+		IssuerRef:             nil,
+		SecretRef:             nil,
+	}
+}
+
+func convertListenerTLSToListenerTLS(name string, tls *ListenerTLS) *redpandav1alpha2.ListenerTLS {
+	if tls == nil {
+		// NB: We have to "explicitly" disable TLS rather than returning nil as
+		// the default helm values might merge in an re-enable TLS. e.g. in the
+		// case of the "default" certs.
+		return &redpandav1alpha2.ListenerTLS{
+			Enabled:           ptr.To(false),
+			Cert:              nil,
+			RequireClientAuth: nil,
+			SecretRef:         nil,
```

Comment:
Looks like this is unused. Was this slurped into `Cert` at some point but never dropped?